app/Http/Controllers/PostController.php

```diff
@@ -0,0 +1,104 @@
+<?php
+
+namespace App\Http\Controllers;
+
+use App\Http\Requests\FormPostRequest;
+use App\Models\Post;
+use Illuminate\Http\Request;
+use Intervention\Image\ImageManagerStatic as Image;
+
+class PostController extends Controller
+{
+    /**
+     * Display a listing of the resource.
+     *
+     * @return \Illuminate\Http\Response
+     */
+    public function index()
+    {
+       $posts =  Post::all();
+        return view('home',['posts'=>$posts]);
+    }
+
+    /**
+     * Show the form for creating a new resource.
+     *
+     * @return \Illuminate\Http\Response
+     */
+    public function create()
+    {
+
+    }
+
+    /**
+     * Store a newly created resource in storage.
+     *
+     * @param  \Illuminate\Http\Request  $request
+     * @return \Illuminate\Http\Response
+     */
+    public function store(FormPostRequest $request)
+    {
+
+        $attributes = $request->validated();
+
+        if(isset($attributes['imagePath']))
+        {
+            $imageName = time() . '-' . $request->input('name') . '-' .  $attributes['imagePath']->getClientOriginalName();
+            $newImage = $attributes['imagePath'];
+            $testImage = Image::make($newImage->getRealpath())->resize(400, 300);
+            $testImage->save(public_path('uploads/' .$imageName));
+            $attributes['imagePath'] = "uploads/$imageName";
+
+        }
+
+
```

Comment:
check out PSR-4 formatting standard and get your IDE to format the code automatically on save.